apps/tangle-dapp/src/pages/restake/withdraw/index.tsx

```diff
@@ -101,29 +102,28 @@ const RestakeWithdrawForm: FC = () => {
   const selectedAsset = useRestakeAsset(selectedAssetId);
 
   const { maxAmount, formattedMaxAmount } = useMemo(() => {
-    if (!delegatorInfo?.deposits) {
+    if (!delegatorInfo?.deposits || selectedAsset === null) {
       return {};
     }
 
-    const depositedAsset = Object.entries(delegatorInfo.deposits).find(
-      ([assetId]) => assetId === selectedAssetId,
+    const availableBalance = calculateRestakeAvailableBalance(
```

Comment:
The backend has a field called [`delegated_amount`](https://github.com/tangle-network/tangle/blob/59c719c14f6f3834546eb22c7b61b119dfd1989b/pallets/multi-asset-delegation/src/types/delegator.rs#L220-L221) that indicates the total delegated amount, which can be used to calculate the available balance. 

This issue has been addressed in PR https://github.com/tangle-network/dapp/pull/2788.